### PR TITLE
add action buttons to the call notification

### PR DESCRIPTION
**Pull request by agrajaghh**
while its ringing:
screenshot_15-44-13

during a call:
screenshot_15-44-24

### CODE REVIEW

**Comment by McLoo** [comment]
fixes #4160
sweet!

**Comment by agrajaghh** [comment]
I was looking for that issue but couldn't find it :grin:

**Comment by moxie0** [comment]
Looking good. Personally I like it when we preserve the original material icon file names, that way it's clear that we don't end up with duplicate resources.

**Comment by agrajaghh** [comment]
thats what I did :stuck_out_tongue: I just removed the suffix (since its not correct):
`ic_phone_48px` -> `ic_phone`
Or would you like to keep it?

**Comment by moxie0** [comment]
The diff I see has added rather than moved files? My assumption was that you copied something like ic_phone_gray600_48dp.png over from material icons.

**Comment by agrajaghh** [comment]
I'm using this [script](https://github.com/sebkur/android-res-utils) to generate the png's from the material icon svg since there is no 32dp version in the icon repo

btw, I used black with 54% opacity. I couldn't find offical info about the color of action icons. is this correct? I just found [this](https://www.google.com/design/spec/style/icons.html#icons-system-icons) (at the bottom)

**Comment by dcherian** [comment]
How about changing the string to "Incoming Signal call" rather than "Signal call in progress" for the first screenshot. Seems weird to say "in progress" when I haven't picked up.

**Comment by agrajaghh** [comment]
@dcherian sounds good, I changed it...

I think it would be even better if the contact name (or number, if no contact) would be displayed. But I'll leave that to another PR..

**Comment by agrajaghh** [comment]
Damm, just realized I forgot outgoing calls. Will fix it soon...

**Comment by jeremymasters** [comment]
@agrajaghh What about a button for denying with text response?  Thought I would ask since Christmas is approaching...;)

**Comment by agrajaghh** [comment]
outgoing calls should be fixed now.

@jeremymasters hm, I think its best to do this seperately, maybe open an issue for that?

outgoing while ringing:
out
I'm not happy with "Signal call in progress". What would be better?

incoming:
inc

established:
established

**Comment by mawed** [comment]
Shouldn't there be a little lock close to the phone to show the encrypted call as it is shown in the conversation wit another Signal user?

**Comment by dcherian** [comment]
Some options for outgoing calls:
 - "Outgoing Signal call"
 - "Establishing Signal call"
 - "Signal: calling NAME_OF_CONTACT"

**Comment by agrajaghh** [comment]
Using "Establishing Signal call" now, but maybe too technical :smirk:
Also using the lock phone icon from the action bar for the notification.

outgoing while ringing:
out

incoming:
inc

established:
established

**Comment by dcherian** [comment]
Does the text string need to be repeated? Could be replaced with contact name in the next PR I guess

**Comment by agrajaghh** [comment]
I thought it's complicated to include the contact names, but it was pretty easy (if I did it correctly :smirk:)

outgoing while ringing:
out

incoming:
inc

established:
established

**Comment by dcherian** [comment]
Looks sweet!

**Comment by mcginty** [comment]
@agrajaghh this looks great! Definitely better with contact names prioritized.

(also for future-times, I found this Android Studio plugin for material asset generation: https://github.com/konifar/android-material-design-icon-generator-plugin)

File: src/org/thoughtcrime/redphone/RedPhone.java [hunk 1]
```diff
@@ -63,6 +63,10 @@
   private static final int STANDARD_DELAY_FINISH    = 1000;
   public  static final int BUSY_SIGNAL_DELAY_FINISH = 5500;
 
+  public static final String ANSWER_ACTION   = RedPhone.class.getName() + ".ANSWER_ACTION";
+  public static final String DENY_ACTION     = RedPhone.class.getName() + ".DENY_ACTION";
+  public static final String END_CALL_ACTION = RedPhone.class.getName() + ".END_CALL_ACTION";
```
Comment (mcginty) [inline src/org/thoughtcrime/redphone/RedPhone.java]: small nitpick: getCanonicalName() is more consistent with how other action strings are constructed in other parts of the app (even though getName() would return the same string in this case)

File: src/org/thoughtcrime/redphone/RedPhone.java [hunk 2]
```diff
@@ -89,6 +93,16 @@ public void onResume() {
     registerBluetoothReceiver();
   }
 
+  @Override
+  public void onNewIntent(Intent intent){
+    if (ANSWER_ACTION.equals(intent.getAction())) {
```
Comment (mcginty) [inline src/org/thoughtcrime/redphone/RedPhone.java]: Java 7 to the rescue! `switch(intent.getAction())`

File: src/org/thoughtcrime/redphone/RedPhone.java (hunk 1 above)
Comment (agrajaghh) [inline src/org/thoughtcrime/redphone/RedPhone.java]: done

File: src/org/thoughtcrime/redphone/RedPhone.java (hunk 2 above)
Comment (agrajaghh) [inline src/org/thoughtcrime/redphone/RedPhone.java]: problem is that `ANSWER_ACTION` etc. are not constant expressions. So either I leave this like it is, or I change the actions to actual strings. What do you prefer?

**Comment by agrajaghh** [comment]
the material asset plugin looks good for the future :blush: But unfortuantely it doesn't support 32dp as well :grin:

**Comment by agrajaghh** [comment]
@mcginty should I include xxxhdpi icons as well? just saw them in #4553...

**Comment by mcginty** [comment]
@agrajaghh yep, we're in the world where xxxhdpi is a thing on new devices now

**Comment by agrajaghh** [comment]
okay, included them now

**Pull request closed without merging**